### PR TITLE
#242 Support footer and header in PoiUtil

### PR DESCRIPTION
**Pull request by SoltauFintel**
First I've tried to extend PoiUtil to copy more page settings.

Then I've added an interface so that JXLS users can change/extend the copy action.

But then I've found `XSSFWorkbook.cloneSheet()` which can do the whole job! So I've changed my interface to SheetCreator. 

You can call `((PoiSheetCreator) builder.getSheetCreator()).setCloneSheet(false)` if you don't want to use cloneSheet(). Or JXLS users can set their own create sheet action using `builder.withSheetCreator(new ...)`.

### CODE REVIEW

File: jxls-site/docs/builder.md [hunk 1]
```diff
@@ -244,6 +244,15 @@ One use case could be performing POI operations, e.g. for grouping.
 If you want to change the way how running variables are accessed, call `withRunVarAccess((name, data) -> your code)`.
 The PublicContext.getRunVar() method is especially used to save loop variables. However, it is often the case that there is no entry for the key at all. If the map implementation reacts allergically to non-existent keys, you can change the behavior with withRunVarAccess().
 
+## Sheet creater
```
Comment (leonate) [inline jxls-site/docs/builder.md:247]: Typo: creater -> creator

File: jxls/src/main/java/org/jxls/builder/SheetCreater.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.jxls.builder;
+
+public interface SheetCreater {
```
Comment (leonate) [inline jxls/src/main/java/org/jxls/builder/SheetCreater.java:3]: I'd change it to `SheetCreator`

**Review by leonate — COMMENTED** [review]
Seems to be a nice change. Besides the spelling "Creater" and missing tests I have no other remarks.

File: jxls-site/docs/builder.md (hunk 1 above)
Comment (SoltauFintel) [reply jxls-site/docs/builder.md]: done

File: jxls/src/main/java/org/jxls/builder/SheetCreater.java (hunk 2 above)
Comment (SoltauFintel) [reply jxls/src/main/java/org/jxls/builder/SheetCreater.java]: done

**Comment by SoltauFintel** [comment]
Development finished. I'm still waiting for the nightly test results.

**Review by leonate — APPROVED** [review]

**Review by leonate — COMMENTED** [review]
Nice work! 👍

**Pull request merged**